nanshe_workflow/data.py

```diff
@@ -23,6 +31,75 @@ def io_remove(name):
         raise ValueError("Unable to remove path, '%s'." % name)
 
 
+def zip_dir(dirname, compression=zipfile.ZIP_STORED, allowZip64=True):
+    filename = dirname + os.path.extsep + "zip"
+    with zipfile.ZipFile(filename, "w", compression, allowZip64) as fh:
+        for root, _, filenames in scandir.walk(os.path.basename(dirname)):
+            for each_filename in filenames:
+                each_filename = os.path.join(root, each_filename)
+                fh.write(
+                    each_filename,
+                    os.path.relpath(each_filename, dirname)
+                )
```

Comment:
We wrote this, which works, but we can actually use the `MutableMapping` interface to drive the conversion as noted in this [comment]( https://github.com/alimanfoo/zarr/issues/137#issuecomment-283405709 ) by using the `update` method. It would be good if we explored this cleaner alternative.